src/transaction/creation_common.rs

```
@@ -2231,12 +2345,21 @@ pub(super) mod package_alias {
 
         let maybe_entity_version = session_version::get(matches);
 
+        let maybe_entity_version_key = if let Some(entity_version) = maybe_entity_version {
+            match major_version::get(matches) {
+                Some(major) => Some(EntityVersionKey::new(major, entity_version)),
```

Not sure if it's an issue but do we care the other way around? If someone specifies major but not entity_version? major would be silently ignored from what I can see. We can also handle this in another PR

---

The issue, is that when specifying a version, we need both parts of the info. So yes, I do agree with the approach!